dipy/denoise/localpca.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import scipy as sp
+
+
+def localpca(arr, sigma, patch_radius=1):
+    r"""Local PCA Based Denoising of Diffusion Datasets
+
+    Parameters
+    ----------
+    arr : A 4D array which is to be denoised
+    sigma : float or 3D array
+        standard deviation of the noise estimated from the data
+    patch_radius : The radius of the local patch to
+                be taken around each voxel
+
+    Returns
+    -------
+    denoised_arr : 4D array
+        this is the denoised array of the same size as that of
+        arr (input data)
+
+    References
+    ----------
+    .. [Manjon13] Manjon JV, Coupe P, Concha L, Buades A, Collins DL
+                  Diffusion Weighted Image Denoising Using Overcomplete Local
+                  PCA, PLOS 2013
+    """
+
+    if arr.ndim == 4:
+
+        tou = 2.3 * 2.3 * sigma
```

Comment:
The sigma from the function is actually sigma squared but yeah I will change the naming so that so that there is no confusion.